content/docs/context.md

```diff
@@ -4,35 +4,71 @@ title: Context
 permalink: docs/context.html
 ---
 
-> Note:
->
-> `React.PropTypes` has moved into a different package since React v15.5. Please use [the `prop-types` library instead](https://www.npmjs.com/package/prop-types) to define `contextTypes`.
->
->We provide [a codemod script](/blog/2017/04/07/react-v15.5.0.html#migrating-from-react.proptypes) to automate the conversion.
+Context provides a way to pass data through the component tree without having to pass props down manually at every level.
+
+## Using Context
+
+Here is an example illustrating how you might inject a "theme" using context:
```

Comment:
I prefer the introductory sequence from the [old "How to Use Context"](https://reactjs.org/docs/context.html#how-to-use-context) section more because of how it setup the problem.

I would suggest following that pattern, but putting it down under the "Typical Usage" header. I think the API should be first on this page.

Reply:
I think it's confusing how the old leads with a large codeblock illustrating non-context APIs. Maybe we can invert the order to highlight the context way before the prop-drilling way?

Moving the API section to the top makes sense.

Reply:
I think we could maybe make the initial code block _shorter_, but I think it's useful to present APIs in the form of "Here is a scenario. How would you solve it? You can solve it using this API."